services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/serialization/ResumableFileUploadSerializer.java

```diff
@@ -0,0 +1,180 @@
+            TransferManagerJsonMarshaller.STRING.marshall(upload.multipartUploadId().get(), jsonGenerator,
+                                                          MULTIPART_UPLOAD_ID);
+        }
+        if (upload.partSizeInBytes().isPresent()) {
+            TransferManagerJsonMarshaller.LONG.marshall(upload.partSizeInBytes().getAsLong(),
+                                                        jsonGenerator,
+                                                        PART_SIZE_IN_BYTES);
+        }
+        if (upload.totalNumOfParts().isPresent()) {
+            TransferManagerJsonMarshaller.LONG.marshall(upload.totalNumOfParts().getAsLong(),
+                                                        jsonGenerator,
+                                                        TOTAL_NUM_OF_PARTS);
+        }
+        marshallUploadFileRequest(upload.uploadFileRequest(), jsonGenerator);
+        jsonGenerator.writeEndObject();
+
+        return jsonGenerator.getBytes();
+    }
+
+    /**
+     * At this point we do not need to persist the TransferRequestOverrideConfiguration, because it only contains listeners and
+     * they are not used in the resume operation.
+     */
+    private static void marshallUploadFileRequest(UploadFileRequest fileRequest, JsonWriter jsonGenerator) {
+        jsonGenerator.writeFieldName(UPLOAD_FILE_REQUEST);
+        jsonGenerator.writeStartObject();
+        jsonGenerator.writeFieldName(SOURCE);
+        jsonGenerator.writeValue(fileRequest.source().toString());
+        marshallPutObjectRequest(fileRequest.putObjectRequest(), jsonGenerator);
+        jsonGenerator.writeEndObject();
+    }
+
+    private static void marshallPutObjectRequest(PutObjectRequest putObjectRequest, JsonWriter jsonGenerator) {
+        jsonGenerator.writeFieldName(PUT_OBJECT_REQUEST);
+        jsonGenerator.writeStartObject();
+        validateNoRequestOverrideConfiguration(putObjectRequest);
+        putObjectRequest.sdkFields().forEach(field -> marshallPojoField(field, putObjectRequest, jsonGenerator));
+        jsonGenerator.writeEndObject();
+    }
+
+    private static void validateNoRequestOverrideConfiguration(PutObjectRequest putObjectRequest) {
+        if (putObjectRequest.overrideConfiguration().isPresent()) {
+            log.debug(() -> "ResumableFileUpload PutObjectRequest contains an override configuration that will not be "
+                            + "serialized");
+        }
+    }
+
+    private static void marshallPojoField(SdkField<?> field, PutObjectRequest request, JsonWriter jsonGenerator) {
+        Object val = field.getValueOrDefault(request);
+        TransferManagerJsonMarshaller<Object> marshaller = getMarshaller(field.marshallingType(), val);
+        marshaller.marshall(val, jsonGenerator, field.locationName());
+    }
+
+    public static ResumableFileUpload fromJson(String bytes) {
+        JsonNodeParser jsonNodeParser = JsonNodeParser.builder().build();
```

Comment:
Here`fromJson` consider Json as String but argument variable name is bytes , should it be named as jsonString ?
 `   public static ResumableFileUpload fromJson(String json) `